pkgs/tools/misc/spigot/default.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, stdenv
+, buildPackages
+, fetchurl
+, gmp
+}:
+
+stdenv.mkDerivation rec {
+  name = "spigot-20200101";
+  src = fetchurl {
+    url = "https://www.chiark.greenend.org.uk/~sgtatham/spigot/${name}.b1b0b20.tar.gz";
```

```suggestion
    url = "https://www.chiark.greenend.org.uk/~sgtatham/spigot/spigot-${version}.b1b0b20.tar.gz";
```

The "b1b0b20" makes me think that this link will probably not remain stable between versions though. Maybe we should use `fetchgit` and fetch from their git repository instead. This will enable automatic updates.

---

> This will enable automatic updates.

In what way are they automatic then?

---

if we come from git, it looks like we'll also need halibut and perl to build the manual

---

Duh, I should have figured this out, but the b1b0b20 is the first part of the git hash

---

I believe for `fetchgit` we need a hash anyways, and the source will then be a fixed-output derivation..?

---

Yeah, I'm convinced.  The current push has fetchgit

---

> > This will enable automatic updates.
> 
> In what way are they automatic then?

[nixpkgs-update](https://github.com/ryantm/nixpkgs-update) will [notice](https://repology.org/project/spigot/versions) when some other package repository has a never version of `spigot`. It will then attempt to update the package by essentially search-and-replacing the old version with the new version (and fixing up fixed-output hashes). If that works, it will create a PR.

That only works when the update is as simple as search-and-replace'ing the version though. If more manual steps are needed, it fails.

---

> if we come from git, it looks like we'll also need halibut and perl to build the manual

My personal preference is to build everything we can from source anyways. Also makes it easier to update to unstable versions / bisect between versions later.

---

> I believe for fetchgit we need a hash anyways, and the source will then be a fixed-output derivation..?

For what its worth, the source is a fixed output derivation whether we use `fetchgit` or `fetchurl` (or any other fetcher).